Patch VS Code to produce a web artifact instead of a desktop application. See:
https://github.com/Felx-B/vscode-web

--- build/gulpfile.vscode.js	2021-08-28 02:52:47.325395558 +0000
+++ build/gulpfile.vscode.js	2021-08-28 02:53:37.461856376 +0000
@@ -35,11 +35,12 @@

 // Build
 const vscodeEntryPoints = _.flatten([
-	buildfile.entrypoint('vs/workbench/workbench.desktop.main'),
+	buildfile.entrypoint('vs/workbench/workbench.web.api'),
 	buildfile.base,
 	buildfile.workerExtensionHost,
 	buildfile.workerNotebook,
-	buildfile.workbenchDesktop,
+	buildfile.workbenchWeb,
+	buildfile.keyboardMaps,
 	buildfile.code
 ]);

@@ -155,8 +156,8 @@

 		const checksums = computeChecksums(out, [
 			'vs/base/parts/sandbox/electron-browser/preload.js',
-			'vs/workbench/workbench.desktop.main.js',
-			'vs/workbench/workbench.desktop.main.css',
+			'vs/workbench/workbench.web.api.js',
+			'vs/workbench/workbench.web.api.css',
 			'vs/workbench/services/extensions/node/extensionHostProcess.js',
 			'vs/code/electron-browser/workbench/workbench.html',
 			'vs/code/electron-browser/workbench/workbench.js'
